rfcs/text/0000-simple-updates.md

```diff
@@ -0,0 +1,159 @@
+
+The wrangler will retrieve a spreadsheet for that submission by clicking a “Download metadata” button or similar on the submission page. The spreadsheet they receive will be the original spreadsheet they submitted plus a column at the end containing the UUIDs of all entities. This column exists only to allow ingest to identify how the rows in the spreadsheet relate to existing metadata when the updated spreadsheet is submitted. It should not be edited by wranglers.
+
+
+* **Step 3 - Wrangler edits the spreadsheet with their change**
+
+The wrangler edits the spreadsheet. They can change any metadata cell apart from the ones containing the UUIDs that ingest added. They can add and remove entries from tabs that contain array/module data. For this solution, they must not add or remove whole entities (e.g. biomaterials, protocols) or change the linking between entities. Both these are instances of experimental design representation changes and are out of scope.
+
+* **Step 4 - Wrangler submits spreadsheet**
+
+The wrangler will submit a spreadsheet containing an update through a “Submit update” button or similar on the submission page. For each submission, we will allow only one update to be submitted and processed at a time. For this solution, the wrangler can only submit an updated spreadsheet that validates against the current schemas. 
+
+* **Step 5 - Ingest shows the diff to the wrangler for approval**
+
+The wrangler will be shown the difference between their updates and the existing submission in the ingest UI. They will also be shown any validation errors. If there are no validation errors then they will have the option either to approve the update, in which case ingest will start submitting it to the datastore, or cancel it, in which case the update will be removed from ingest and they can return to step 3. If there are validation errors then the wrangler can only cancel the update and submit a new update spreadsheet.
+
+* **Step 6 - Ingest uploads new versions of metadata files and bundles to the datastore**
+
+For this solution, ingest will calculate the bundle updates and submit these to the datastore. This represents a risk to the scalability of this solution. If performance problems emerge we will need to consider solutions in a future RFC.
+
+### DCP Flow of Updates
+
+![](../images/0000-simple-updates-dcp-flow.png)
+
+Once the bundles are updated (1 & 2), the datastore will send update notifications to all listeners (3). The query service, data browser and matrix service **MUST** all process these notifications to provide the updated metadata to consumers as required. The data processing pipeline execution service **MUST NOT** submit new or updated analyses to ingest for assays that have previously had analysis submitted. This solution requires the data processing pipeline execution service to capture and check if notification has already triggered a pipeline run and stop the run if it has.
```

Comment:
Yes, I believe planners deliberately restricted updates scope to avoid having to decide on these questions right now, which were generating a lot of discussion without obvious conclusions. @lauraclarke, @morrisonnorman could you make sure this limitation is explicit?

Regarding your comment:

> I think it's also worth spelling out what happens if a new analysis bundle is submitted in this case. A simple solution would be to retain the output bundle UUID, but completely replace its contents with new outputs.

This is indeed one solution I was contemplating. Didn't explore it further once the scope was restricted. It may avoid the need to make various decisions about analysis updates in section c) in a previous comment.

Reply:
> Yes, this was one solution I was thinking about before the Q2 scope restriction

This discussion seems reasonable given the scope restriction. I want to flag, for the future, that it is likely that analysis pipelines will consume metadata, such as batch, replicate, or patient information, to carry out normalization. In these cases, changes to certain metadata fields **MUST** provoke reanalysis.

Reply:
@ambrosejcarr We are in full agreement that this is functionality which will be required. We decided that figuring out the logic for reanalysis vs analysing with a new analysis pipeline (and if the old analysis results do or don't need to be reacted) was sufficiently complex that we didn't want it to delay the simple updates.  We must make sure that conversation starts and we can get an RFC about the plan for solving that problem.

Reply:
Just to clarify `The data processing pipeline execution service **MUST NOT** submit new or updated analyses to ingest for assays that have previously had analysis submitted.` If an update comes in and only the pipeline version has changed, is the expectation here that we would not re-analyze the data with the updated pipeline until a mechanism is in place to handle re-analysis?

Reply:
Yep